extensions/oidc-client-registration/deployment/src/main/java/io/quarkus/oidc/client/registration/deployment/devservices/keycloak/KeycloakDevServiceRequiredBuildStep.java

```diff
@@ -0,0 +1,93 @@
+package io.quarkus.oidc.client.registration.deployment.devservices.keycloak;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.logging.Logger;
+import org.keycloak.common.util.MultivaluedHashMap;
+import org.keycloak.representations.idm.ComponentExportRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.BuildSteps;
+import io.quarkus.deployment.builditem.DockerStatusBuildItem;
+import io.quarkus.deployment.dev.devservices.GlobalDevServicesConfig;
+import io.quarkus.devservices.keycloak.KeycloakAdminPageBuildItem;
+import io.quarkus.devservices.keycloak.KeycloakDevServicesConfigurator;
+import io.quarkus.devservices.keycloak.KeycloakDevServicesRequiredBuildItem;
+import io.quarkus.devui.spi.page.CardPageBuildItem;
+import io.quarkus.oidc.client.registration.deployment.OidcClientRegistrationBuildStep;
+import io.quarkus.runtime.configuration.ConfigUtils;
+
+@BuildSteps(onlyIfNot = IsNormal.class, onlyIf = { OidcClientRegistrationBuildStep.IsEnabled.class,
+        GlobalDevServicesConfig.Enabled.class })
+public class KeycloakDevServiceRequiredBuildStep {
+
+    private static final Logger LOG = Logger.getLogger(KeycloakDevServiceRequiredBuildStep.class);
+    private static final String OIDC_CLIENT_REG_AUTH_SERVER_URL_CONFIG_KEY = "quarkus.oidc-client-registration.auth-server-url";
+    private static final String OIDC_AUTH_SERVER_URL_CONFIG_KEY = "quarkus.oidc.auth-server-url";
+
+    @BuildStep
+    void requireKeycloakDevService(BuildProducer<KeycloakDevServicesRequiredBuildItem> keycloakDevSvcRequiredProducer,
+            DockerStatusBuildItem dockerStatusBuildItem) {
+        if (ConfigUtils.isPropertyNonEmpty(OIDC_CLIENT_REG_AUTH_SERVER_URL_CONFIG_KEY)) {
+            LOG.debugf("Not starting Dev Services for Keycloak as '%s' has been provided",
+                    OIDC_CLIENT_REG_AUTH_SERVER_URL_CONFIG_KEY);
+            return;
+        }
+        // this is for backwards compatibility - when users configured named client with OIDC auth server URL
+        // that is not our Keycloak Dev Service, we don't want to start our Dev Service to keep previous behavior
+        if (ConfigUtils.isPropertyNonEmpty(OIDC_AUTH_SERVER_URL_CONFIG_KEY)) {
+            LOG.debugf("Not starting Dev Services for Keycloak as '%s' has been provided",
+                    OIDC_AUTH_SERVER_URL_CONFIG_KEY);
+            return;
+        }
```

Comment:
What happens if only `quarkus-oidc-client-registration` and `quarkus-oidc-client` are loaded ? Is Keycloak launched twice ?

Reply:
I think I can put it even simpler: would we start KC Dev Svc for OIDC Client (/registration) if OIDC is using different authentication server? My answer is no, but if the answer is yes, we will need to tweak DEV UI because currently it doesn't support this situation.